src/app.test.js

```diff
@@ -180,16 +202,32 @@ test("shows outstanding todos when filter is set to Outstanding", () => {
   expect(queryByDisplayValue(todoList, "Take dog for a walk!")).toBeVisible();
 });
 
-/**
- * SETUP
- * render todo app
- * input user name
- * add few todos
- * destroy todo app
- * ACT
- * re render todo app
- * input same user name
- * ASSERT
- * see previously added todos rendered
- */
-test.todo("renders todo list for existing user");
+test("renders empty todo list when changing to a nonexistent user", () => {
+  // setup
+  logInUser();
+  const todoList = screen.getByTestId("todo-list");
+  addTodo(todoList, "Take wife for a walk!", true);
+  addTodo(todoList, "Take dog for a walk!");
+
+  // act
+  logInUser("Marton");
+
+  // assert
+  expect(todoList).toBeEmptyDOMElement();
+});
+
+test("renders todo list for existing user", () => {
+  // setup
+  logInUser();
+  const todoList = screen.getByTestId("todo-list");
+  addTodo(todoList, "Take wife for a walk!", true);
+  addTodo(todoList, "Take dog for a walk!");
+
+  // act
+  logInUser("Marton");
+  logInUser();
```

Comment:
this is the case where it would be better to _explicitly_ give the username to `logInUser`, otherwise the reader can get confused and forced to go and check what's happening in `logInUser` just to find out there's a default username "Attila"

for demonstration we could just use "A" and "B" to improve comprehensiveness

also, it can be quite difficult to distinguish between "Take wife for a walk!" and "Take dog for a walk!" as they differ only in a few characters, would be much better to use e.g. "Todo 1 of user A", "Todo 2 of user A" etc. and "Todo 1 of user B", "Todo 2 of user B" respectively

Reply:
actually, thinking of this "log in", it is essentially a "change/switch user" action rather than a "log in" action, give it a try and rename it to see if that reads better in the test cases e.g. `changeUserTo("A")` or `switchUserTo("B")`